type.go

```diff
@@ -1722,11 +1722,45 @@ func (t *timestampType) Decode(dst encoding.Values, src []byte, enc encoding.Enc
 }
 
 func (t *timestampType) AssignValue(dst reflect.Value, src Value) error {
-	return Int64Type.AssignValue(dst, src)
+	switch dst.Type() {
+	case reflect.TypeOf(time.Time{}):
+		// TODO: Need to figure out the source unit rather than assuming nanos
```

Comment:
The source unit is the unit of `t` isn't it?

Reply:
Indeed it is 👍